actionview/lib/action_view/template.rb

```diff
@@ -324,8 +325,11 @@ def handle_render_error(view, e) #:nodoc:
       end
 
       def locals_code #:nodoc:
+        locals = @locals.to_set - Module::DELEGATION_RESERVED_METHOD_NAMES
+        locals = locals.select { |key| /\A[a-z_][a-zA-Z_0-9]*\z/.match?(key) }
```

Comment:
We need to require the match? core ext file that backports it all the way to Ruby 2.2.